extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java

```diff
@@ -239,45 +238,43 @@ private boolean isBinary(String contentType) {
     }
 
     private Optional<Principal> getPrincipal(APIGatewayV2HTTPEvent request) {
-        if (ImageInfo.inImageRuntimeCode()) {
-            return Optional
-                    .ofNullable(request.getRequestContext())
-                    .map(APIGatewayV2HTTPEvent.RequestContext::getAuthorizer)
-                    .map(APIGatewayV2HTTPEvent.RequestContext.Authorizer::getJwt)
-                    .map(APIGatewayV2HTTPEvent.RequestContext.Authorizer.JWT::getClaims)
-                    .map(claims -> {
-                        final String jwtUsername = claims.get("cognito:username");
-                        return new Principal() {
-                            @Override
-                            public String getName() {
-                                return jwtUsername;
-                            }
-                        };
-                    });
-        } else {
-            // we might want to hack the user, but only if:
-            // AWS_SAM_LOCAL == true
-            // QUARKUS_LAMBDA_FORCE_USER == true
-            final Map<String, String> systemEnvironment = System.getenv();
-            if (systemEnvironment != null) {
-                final boolean isSamLocal = Boolean.parseBoolean(systemEnvironment.get("AWS_SAM_LOCAL"));
-                final boolean isForceLocalUser = Boolean.parseBoolean(systemEnvironment.get("QUARKUS_AWS_LAMBDA_FORCE_USER"));
+        final Map<String, String> systemEnvironment = System.getenv();
+        final boolean isSamLocal = Boolean.parseBoolean(systemEnvironment.get("AWS_SAM_LOCAL"));
```

Comment:
Nice, even better.  You can detect whether you are running with SAM LOCAL?

Reply:
see my previous comment above, but do we even need this AWS_SAM_LOCAL env var ? If I provide the FORCE_USERNAME then it should be enough in any env (local, jvm, native).